Fix SpankBang uploader and uploader_id

f4pp3rk1ng opened the pull request:
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Fixes https://github.com/yt-dlp/yt-dlp/issues/833 by extracting the correct `uploader`.
Also adds the `uploader_id`.

coletdjnz commented on yt_dlp/extractor/spankbang.py:
```diff
@@ -140,9 +140,9 @@ def extract_format(format_id, format_url):
-            webpage, 'uploader', default=None)
+            r'</use></svg> (.*) <svg class="i_svg i_angle-right">', webpage, 'uploader_id', default=None)
+        uploader_id = self._html_search_regex(
+            r'<a href="/profile/(.*)" class="ul">', webpage, 'uploader', default=None)
```
```suggestion
            r'<a[^>]+href="/profile/([^"]+)"', webpage, 'uploader_id', default=None)
```

coletdjnz commented on yt_dlp/extractor/spankbang.py:
```diff
@@ -140,9 +140,9 @@ def extract_format(format_id, format_url):
-            (r'(?s)<li[^>]+class=["\']profile[^>]+>(.+?)</a>',
-             r'class="user"[^>]*><img[^>]+>([^<]+)'),
-            webpage, 'uploader', default=None)
+            r'</use></svg> (.*) <svg class="i_svg i_angle-right">', webpage, 'uploader_id', default=None)
```
```suggestion
            r'<svg[^>]+\bclass="(?:[^"]*?user[^"]*?)">.*?</svg>([^<]+)', webpage, 'uploader', default=None)
```

coletdjnz commented on yt_dlp/extractor/spankbang.py:
```diff
@@ -157,6 +157,7 @@ def extract_format(format_id, format_url):
             'description': description,
             'thumbnail': thumbnail,
             'uploader': uploader,
+            'uploader_id': uploader_id,
```
Add a test containing the new field. The current test looks dead, so you can probably replace it with a new one.

coletdjnz reviewed (changes requested):
Your regex needed relaxing a bit. I've had a go at improving it, @pukkandan may want to have a quick look over.
https://github.com/ytdl-org/youtube-dl#make-regular-expressions-relaxed-and-flexible

Relevant data we are looking into:

```
<a href="/profile/myuploaderid" class="ul"><svg class="username"><use xlink:href="/static/desktop/gen/universal.master.5.56.d17633a5.svg#user"></use></svg> MyUploader <svg class="i_svg i_angle-right"><use xlink:href="/static/desktop/gen/universal.master.5.56.d17633a5.svg#angle-right"></use></svg></a>
```

pukkandan reviewed (approved):
The current regexes look good. Can be merged once test is added

f4pp3rk1ng commented:
I was unable to get the tests to run but I fixed the test and checked the correctness manually.
I also corrected the title regex since it was including all the tags (which SpankBang nests in the `h1` element).

pukkandan commented:
> I was unable to get the tests to run 

wdym?

f4pp3rk1ng commented:
> > I was unable to get the tests to run
> 
> wdym?

I could not get Python/the unit testing framework to execute my unit test.
I manually checked the results so they should pass but I couldn't confirm it.

I ran `python test/test_download.py TestDownload.test_spankbang` as per point 6 of https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site but it didn't work.
